pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -618,14 +618,18 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
     updateCurrentDocumentCountMetrics();
     if (messagesAndOffsets.getUnfilteredMessageCount() > 0) {
       _hasMessagesFetched = true;
+      if (messageCount == 0) {
+        // If we did not get any events but got some unfiltered messages, we attempt to estimate the ingestion
```

Comment:
can we rephrase this comment as it is super confusing:
`If we received events from stream but all were filtered, we attempt to ...` ?

better yet,  use `batch.getMessageCount() == 0`

Reply:
yes, there was some wording issue but now I have updated it correctly, as for the last suggestion they are it is the same right, not sure what advantage would be to call the method again here if we have the outcome in messageCount other than the overhead of the call?

Reply:
I don't think the ovehead is a lot and it definitely trumps the readability of the code. There are 3-4 types of counts that we track here in a very C style code. Hence, the suggetion

Reply:
Since the variable is used in many other places I will keep using it, may be change it in a future PR where we replace all uses for method invocation